src/core/data-structures/curve.js

```diff
@@ -16,12 +16,12 @@
 
 // Imported just for the side effect of getting the `types` it exports into
 // the type system during compile time.
-import './time';
+import {NormTimeDef} from '../types/date';
 
 /**
  * A CurveDef is a function that returns a normtime value (0 to 1) for another
  * normtime value.
- * @typedef {function(./time.normtimeDef): ./time.normtimeDef}
+ * @typedef {function(NormTimeDef): NormTimeDef}
  */
 export let CurveDef;
```

Comment:
The calls to `bezierCurve` inside `Curves` const are not pure. Please define them as methods that call into cached bezier curves.

Reply:
`once` is a cool helper https://github.com/ampproject/amphtml/pull/34253/commits/48d910d7e3aff7d2ff22f462b0d28e4e57019f5f

Reply:
That actually doesn't help, because `once` isn't pure either! lol.

The `core` directory needs to be completely static evaluable for definitions. If possible, it might be a good idea to rewrite these to call directly into a top-level implementation of `solveYValueFromXValue`.

Reply:
I'm just going to make the whole `Bezier` class static then

Reply:
@jridgewell Done, PTAL when convenient